cantools/database/can/message.py

```diff
@@ -349,6 +351,27 @@ def get_prefix(index, length):
         def add_prefix(prefix, lines):
             return [prefix + line for line in lines]
 
+        def format_signal_line(signal_name):
```

Comment:
hm, could the more sophisticated signal formatting stuff be moved to the `dump.utility` or similar? to me this looks too much tailored for terminal output while the `database` submodule ought to be about the bus description...

Reply:
Hm, yes, I agree with that. The affected interface in cantools.database.can.message.Message is signal_tree_string() which is only used by dump. Shall I move that entire method to dump instead?

Reply:
> Shall I move that entire method to dump instead? That is true for many formatting methods in the Message class, maybe put them all in a dump.utility module?

good idea. As I see it the formatting stuff in `database/` should be limited to a a no-frills string representation of the messages, signals, etc. Fancier formatting could be provided by a utility module like you propose.

> If needed elsewhere in the future, we can just move that module to a more suitable place?

+1

Reply:
I created subparsers.dump.formatting and moved all dump-only formatting stuff into that, and also all relevant test cases. The commit(s) became pretty large in total, but I think that the message module looks better now.